### ISSUE TITLE
list_applications method fail when a Service application is returned

### ISSUE THREAD

**Issue by RomainMichau** [okta/okta-sdk-python#287]
# Context:
I created an API Service APP via Okta Admin console folllowing the [okta doc.](https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/main/#generate-the-jwk-in-the-admin-console) This application seems to have a DYNAMIC issuer_mode (I did not set it myself, it was setted by default)
When i try to get a list of my applications using the python SDK I get an error message `'DYNAMIC' is not a valid OpenIdConnectApplicationIssuerMode`. Although it seems to be a valid since it was setted by my okta tenant.
This issue prevent me to list my applications

# Code:
```python
config = {
            'orgUrl': org_url,
            'authorizationMode': 'PrivateKey',
            'clientId': service_app_client_id,
            'scopes': ['okta.apps.read'],
            'privateKey': jwk,  
            "logging": {"enabled": True, "logLevel": logging.DEBUG}
}
okta_client = OktaClient(config)
apps, res, error  = await okta_client.list_applications({"filter" :  f"name eq \"oidc_client\""})
res._body[0]['settings']['oauthClient']['issuer_mode'])
# DYNAMIC
print(apps)
# None
print(error)
# 'DYNAMIC' is not a valid OpenIdConnectApplicationIssuerMode
```

**Comment by MasterSergius** [comment]
@RomainMichau Looks like the issue is the same as this one: https://github.com/okta/okta-sdk-python/issues/286
According to my investigation, we need to add `DYNAMIC` issuer mode to openapi spec first https://github.com/okta/okta-management-openapi-spec, then we can re-generate python SDK. Will keep you posted.

**Comment by Interperle** [comment]
when is #291 merged to solve this issue? I really need this fix.

**Comment by RomainMichau** [comment]
@Interperle 
FYI as a temporary fix I did that: 
```
        apps, res, error = await self.__okta_client.list_applications({"filter": f"name eq \"oidc_client\""})
        try:
            return [app for app in res._body]
        except IndexError:
            return None
```
Basically you can get the list of app in the res object

**Comment by drewcarmichael-okta** [comment]
Fixed in https://github.com/okta/okta-sdk-python/pull/313

**Comment by explody** [comment]
It looks like this was fixed in 15b2a9986c5746bbacf387a75bf6a9b9010171c9 and then broken again in 15b2a9986c5746bbacf387a75bf6a9b9010171c9 (CC @justinabrokwah-okta).

**Comment by bretterer** [comment]
Resolved in 2.9.0
